streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -93,7 +95,7 @@
      *          |      +-----+-------+      |
      *          +<---- | Partitions  |      |
      *          |      | Assigned (3)| <----+
-     *          |      +-----+-------+      |
+     *          |      +-----+-------+      ^
```

Comment:
Before this change a transition from `STARTING` to `RUNNING` was possible in the diagram. However, such a transition is not valid.

Reply:
This seems a bit subtle. What if we instead break up the loop from Running --> Running below?

Reply:
I had the same feeling. I will change it.